packages/modules/manage-versions/src/components/ManageVersions/ManageVersions.tsx

```diff
@@ -164,23 +197,75 @@ export const ManageVersions = (props: ManageVersionsProps) => {
     getVersions();
   }, [getVersions]);
 
+  const updateProperties = React.useCallback(() => {
+    if (versions) {
+      setVersions((prevVersions) =>
+        (prevVersions ?? []).map((newVersion) => {
+          const creatorId = newVersion._links.creator.href.substring(
+            newVersion._links.creator.href.lastIndexOf("/") + 1
+          );
+          return {
+            ...newVersion,
+            createdBy: usersRef.current ? usersRef.current[creatorId] : "",
+          };
+        })
+      );
+    }
+    if (changesets) {
+      setChangesets((prevChangesets) =>
+        (prevChangesets ?? []).map((changeSet) => ({
+          ...changeSet,
+          createdBy: usersRef.current
+            ? usersRef.current[changeSet.creatorId]
+            : "",
+        }))
+      );
+    }
+  }, [changesets, versions]);
+
   React.useEffect(() => {
     if (versionStatus === RequestStatus.NotStarted) {
       getVersions();
     }
   }, [getVersions, versionStatus]);
 
+  React.useEffect(() => {
+    const loadUsers = async () => {
+      await getUsers();
+    };
+    if (!usersRef.current) {
+      loadUsers()
+        .then(() => {
+          setShouldUpdateProperties(true);
```

Comment:
why you need this additional state?
It looks hacky..
Cant you just run a function to remap the data...?
Also, have separate functions for NV and Changesets.
And what happens if users request finish first?

Reply:
To update version and changeset states with "created By " property we need to ensure users request is finish then only update states, that's why maintaining additional state "shouldUpdateProperties"

Reply:
created separate func for nv and changeset.
removed state and instead calling those function to update state